src/js/ParticleSystem.js

```diff
@@ -0,0 +1,123 @@
+import THREE from 'three';
+
+function applyPointRandomness(r) {
+  let p = new THREE.Vector3(
+    (Math.random() - 0.5),
+    (Math.random() - 0.5),
+    (Math.random() - 0.5))
+    .normalize()
+    .multiplyScalar(r);
+  return p;
+}
+
+function newParticlePosition(oldPosition, newPosition, lerpFactor, offset, pointRandomness) {
```

Comment:
Why not just use emitter and lerpFactor as arguments? Now the code for calculating the arguments is almost identical.